### DISCUSSION TITLE
Change the way Rectangle adjust window's size and position.

### DISCUSSION THREAD

**Discussion by hiangeel** [rxhanson/Rectangle#665] [Ideas]
Hi,
I have just used Rectangle but I love it so much. Thanks for an amazing utility.
I have a suggestion. Currently, we have different shortcuts for arrange windows at half and quarter of screen. There is no problem with `Ctr + Option + Arrow key` for half screen arrangement, but the corner arrangement's shortcuts seem is not convenient. 
My idea is continue using `Ctr + Option + Arrow key` shortcut to switch between half and corner arrangement. For example, if my window's app is on the top half, I will use `Ctr + Option + Left` keys to move to top left quarter. Then If I want to bring it to the half left, I use `Ctr + Option + Down` keys.
Hope you understand what I said.

**Comment by rxhanson** [comment]
Sounds like you are looking for something similar to Windows? I don’t have plans to add that into Rectangle at this time, but would be fine merging a good PR for it, having it be something configurable via Terminal command. 

I did add the Windows style state machine in my paid superset of Rectangle, Hookshot (https://hookshot.app). Hookshot had some necessary components for the feature already in, which made it simple to add there while it is not as simple to add it to Rectangle.

**Reply by hiangeel** [reply]
I'm not sure about that feature on Windows. Anyway, I am not familiar with "PR" that you was talking about. What is it?

**Reply by rxhanson** [reply]
PR: pull request. If you want to implement your desired functionality then I would consider merging your implementation in if it looks good. As mentioned, it’s not as trivial as it sounds and I would recommend just trying Hookshot and configuring the “win arrow keys” in the keyboard shortcuts there and seeing if that comes close enough to what you want.

**Reply by hiangeel** [reply]
Thanks for your response. I'll take a look at Hookshot

**Comment by X-Wei** [comment]
I implemented this — PR #1824.

With the `halvesPreserveOtherAxisSize` default enabled, Left/Right/Top/Bottom Half only change their own axis and keep the window's size on the other one, so Top Half then Left Half gives the top left quarter, and Bottom Half from there expands it back to Left Half (Right Half would give Top Half). Untiled windows and repeated executions of a plain half behave exactly as today, so the repeated-command setting is unaffected.

It is off by default and configurable via Terminal command (`defaults write com.knollsoft.Rectangle halvesPreserveOtherAxisSize -bool true`, documented in TerminalCommands.md); the PR also adds a Settings checkbox, which can be dropped if you'd rather keep it terminal-only. The logic lives in one self-contained class with 32 unit tests, and there is a notarized test build in the PR description.
